geniza/footnotes/models.py

```diff
@@ -410,6 +411,43 @@ def editions(self):
         return self.filter(doc_relation__contains=Footnote.EDITION)
 
 
+class HTMLLineNumberParser(HTMLParser):
+    """HTML parser to add numbering to line elements for search indexing purposes"""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize empty string and line numbering at 1"""
+        self.html_str = ""
+        self.line_number = 1
+        super().__init__(*args, **kwargs)
+
+    def handle_starttag(self, tag, attrs):
+        """Restart line numbering on <ol>, include line number with <li>, and construct
+        start tags with included attributes"""
+        if tag == "ol":
+            # restart line numbering on encountering ol
+            self.line_number = 1
+            for (attr, val) in attrs:
+                # if start present in attrs, restart to start number
+                if attr == "start":
+                    self.line_number = int(val)
+        elif tag == "li":
```

Comment:
I think we need a state indicator that we are within an `ol`, because that's the only time this logic should happen. (Possible we could have a `ul`)